l2geth/rollup/fees/rollup_fee_test.go

```diff
@@ -102,3 +103,80 @@ func TestCalculateRollupFee(t *testing.T) {
 		})
 	}
 }
+
+func TestPaysEnough(t *testing.T) {
+	tests := map[string]struct {
+		opts *PaysEnoughOpts
+		err  error
+	}{
+		"missing-gas-price": {
+			opts: &PaysEnoughOpts{
+				GasLimit:      0,
+				GasPrice:      nil,
+				Fee:           new(big.Int),
+				ThresholdUp:   nil,
+				ThresholdDown: nil,
+			},
+			err: errMissingInput,
+		},
+		"missing-fee": {
+			opts: &PaysEnoughOpts{
+				GasLimit:      0,
+				GasPrice:      new(big.Int),
+				Fee:           nil,
+				ThresholdUp:   nil,
+				ThresholdDown: nil,
+			},
+			err: errMissingInput,
+		},
+		"equal-fee": {
+			opts: &PaysEnoughOpts{
+				GasLimit:      1,
+				GasPrice:      BigTxGasPrice,
+				Fee:           new(big.Int).SetUint64(1),
+				ThresholdUp:   nil,
+				ThresholdDown: nil,
+			},
+			err: nil,
+		},
+		"fee-too-low": {
+			opts: &PaysEnoughOpts{
+				GasLimit:      1,
+				GasPrice:      BigTxGasPrice,
+				Fee:           new(big.Int).SetUint64(2),
+				ThresholdUp:   nil,
+				ThresholdDown: nil,
+			},
+			err: errFeeTooLow,
+		},
+		"fee-threshold-down": {
+			opts: &PaysEnoughOpts{
+				GasLimit:      1,
+				GasPrice:      BigTxGasPrice,
+				Fee:           new(big.Int).SetUint64(2),
+				ThresholdUp:   nil,
+				ThresholdDown: new(big.Float).SetFloat64(0.5),
+			},
+			err: nil,
+		},
```

Comment:
This test is confusing me -- what is `BigTxGasPrice `? It must be 1 for this test to work?

Reply:
Ah clarified with Mark -- the "fee" was really the "expected gas limit" because the compare fee was multiplying the "fee" by the gas price.

Reply:
The values in the structs have been updated, let me know if they are unclear still

Reply:
Way cleaner